ft/ft/functions.py

```diff
@@ -376,3 +376,14 @@ def less_than(i, inp):
 def less_equals(i, inp):
     i = dynamic_cast(T_INT, i).value
     return inp <= i
+
+
+@register("format")
+@typed(None, T_STRING)
+def format(format_string, inp):
+    try:
+        return (format_string.value).format(float(inp))
```

Comment:
are these parens necessary?

Why cast the `inp`ut to `float`? Wouldn't this also work without the cast?

Reply:
Parens not necessary, thanks!
As for the cast, it seems like it doesn't work, actually! I haven't looked at the `dynamic_cast` function, so is that something that could work?

Reply:
One way to do it is to check if the `format_string` ends with an `f` (float) or a `d` (decimal integer), and if so cast the input to the corresponding type. It might become somewhat messy, though.

Reply:
The problem is that we don't have `float` as a type in shell-functools. If you remove the `float(..)` cast, everything works fine for strings and integers, for example:

``` python
def test_map_format_string():
    map_test = mock_command(Map)
    map_test.set_input("format", ["{:>5}"], ["abc", "b"])
    map_test.assert_output(["  abc", "    b"])


def test_map_format_int():
    map_test = mock_command(Map)
    map_test.set_input("format", ["{:02x}"], ["3", "11", "255"])
    map_test.assert_output(["03", "0b", "ff"])

```

I think we should leave it without the float cast at the moment and think about another way to integrate floats later (add it as an official type in shell-functools, add a cast-function that can be called via `map cast float` or similar, ...).